ArduPlane/Log.cpp

```diff
@@ -245,6 +257,31 @@ void Plane::Log_Write_RC(void)
     Log_Write_AETR();
 }
 
+void Plane::Log_Write_Guided(void)
+{
+#if OFFBOARD_GUIDED == ENABLED
+    if (control_mode != &mode_guided) {
+        return;
+    }
+
+    if (guided_state.target_heading_time_ms != 0) {
+        logger.Write_PID(LOG_PIDG_MSG, g2.guidedHeading.get_pid_info());
+    }
+
+    if (guided_state.target_alt > 0.0f || (guided_state.target_airspeed_cm > 0.001f )) {
+        logger.Write("OFG", "TimeUS,Arsp,ArspA,Alt,AltA,AltF,Hdg,HdgA", "QffffBff",
```

Comment:
Why are you logging adhoc like this if you declared the struct above? Struct based logging is more efficent, but we should at least pick just one.

Reply:
fixed on next attempt.